fedmsg/consumers/__init__.py

```diff
@@ -226,11 +227,30 @@ def _make_query(page=1):
                         break
 
     def validate(self, message):
-        """ This needs to raise an exception, caught by moksha. """
+        """
+        Validate the message before the consumer processes it.
+
+        This needs to raise an exception, caught by moksha.
+
+        Args:
+            message (dict): The message as a dictionary. This must, at a minimum,
+                contain the 'topic' key with a unicode string value and 'body' key
+                with a dictionary value.
+
+        Raises:
+            RuntimeWarning: If the message is not valid.
+        """
         if hasattr(message, '__json__'):
             message = message.__json__()
-            if isinstance(message['body'], str):
+            if isinstance(message['body'], six.text_type):
                 message['body'] = json.loads(message['body'])
+            elif isinstance(message['body'], six.binary_type):
+                # Try to decode the message body as UTF-8 since it's very likely
+                # that that was the encoding used. This API should eventually only
+                # accept unicode strings inside messages. If a UnicodeDecodeError
+                # happens, let that bubble up.
+                self.log.error('Message body is not unicode; this is deprecated')
```

Comment:
I recommend using the ```warnings``` module here to issue a deprecation warning, rather than logging an error. After all, using deprecated code isn't actually an error… yet ☺